### ISSUE TITLE
Sik-Ho Tang | Brief Review -- Distilling Visual Priors from Self-Supervised Learning.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#143]
Sik-Ho Tang. [Brief Review — Distilling Visual Priors from Self-Supervised Learning](https://sh-tsang.medium.com/brief-review-distilling-visual-priors-from-self-supervised-learning-e8377118e797).

**Comment by NorbertZheng** [comment]
## Overview
Using [MoCo v2](https://sh-tsang.medium.com/review-moco-v2-improved-baselines-with-momentum-contrastive-learning-8b03598a0324) as Teacher, Knowledge [Distillation](https://sh-tsang.medium.com/review-model-distillation-distilling-the-knowledge-in-a-neural-network-image-classification-48ce0c81618a) for Student, in VIPriors Challenge.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/90e880e6-72a9-4968-8813-5f2cae979723)
VIPriors Challenge (Image from [2020 ECCV Workshop VIPriors Challenge](https://competitions.codalab.org/competitions/23713)).

[Distilling Visual Priors from Self-Supervised Learning](https://link.springer.com/chapter/10.1007/978-3-030-66096-3_29)
MoCo v2+Distillation, by Tongji University, and Megvii Research Nanjing
2020 ECCV Workshop VIPriors Challenge

 - This is a paper participating in the [“Visual Inductive Priors for Data-Efficient Computer Vision” Challenges (VIPriors) in 2020 ECCV Workshop](https://competitions.codalab.org/competitions/23713), which is under the data-deficient scenario.
 - The first phase is to learn a teacher model using [MoCo v2](https://sh-tsang.medium.com/review-moco-v2-improved-baselines-with-momentum-contrastive-learning-8b03598a0324).
 - The second phase is to distill the representations into a student model in a self-[Distillation](https://sh-tsang.medium.com/review-model-distillation-distilling-the-knowledge-in-a-neural-network-image-classification-48ce0c81618a) manner.

**Comment by NorbertZheng** [comment]
## Proposed Framework
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/43e1b74a-c7b7-41c9-a7ac-a95dc321feed)
Proposed Framework.

There are 2 phases. Phase-1 for teacher and Phase-2 for student.

**Comment by NorbertZheng** [comment]
### Phase 1: Teacher
 - [MoCo v2](https://sh-tsang.medium.com/review-moco-v2-improved-baselines-with-momentum-contrastive-learning-8b03598a0324) is used to train the backbone in self-supervised manner for 800 epochs (best model obtained from MoCo v2).
 - The original loss used by [MoCo v2](https://sh-tsang.medium.com/review-moco-v2-improved-baselines-with-momentum-contrastive-learning-8b03598a0324) is:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/e8bee94f-6858-4c14-b391-4f05fe6f4370)
 - [MoCo v2](https://sh-tsang.medium.com/review-moco-v2-improved-baselines-with-momentum-contrastive-learning-8b03598a0324) uses a momentum encoder $\theta_{k}$ to encode all the $k$ and put them in a queue for negative sampling. The momentum encoder is a momentum average of the encoder $\theta_{q}$:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/70f72f8e-68df-4726-b32b-c90f723383a2)

In a data-deficient dataset, the maximum size of the queue is limited, authors propose to
 - add a margin to the original loss function to help the model obtain a larger margin between data samples thus help the model obtain a similar result with fewer negative examples:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/4e3afe7f-1ed6-416e-bb0d-9e4ebeb594f7)
queue size is 4096.

**Comment by NorbertZheng** [comment]
### Phase-2: Self-Distillation on Labeled Dataset
The [Distillation](https://sh-tsang.medium.com/review-model-distillation-distilling-the-knowledge-in-a-neural-network-image-classification-48ce0c81618a) process can be seen as a regulation to
 - **prevent the student from overfitting the small train dataset**,
 - and **give the student a more diverse representation for classification**.

Following OFD, the [Distillation](https://sh-tsang.medium.com/review-model-distillation-distilling-the-knowledge-in-a-neural-network-image-classification-48ce0c81618a) loss is:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/fcfb5eee-1a38-4957-8f49-b4cd35433605)
where distance metric $d_{p}$ is l2-distance in this paper.

Along with a cross-entropy loss for classification:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/68f00c2c-7152-4046-88bc-01b3e2206e3c)

The final loss function for the student model is:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/cae236cf-d8ee-464e-9bdd-1f5783debb6c)
$\lambda=10^{-4}$. 100 epochs are used for fine-tuning.

**Comment by NorbertZheng** [comment]
## Results
## Dataset
 - A small subset of ImageNet is used as the [VIPrior challenge](https://competitions.codalab.org/competitions/23713) dataset.

>There are still 1,000 classes but 50 images for each class in each train/val/test split, resulting in a total of 150,000 images.

### Performance
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/fd1f9cf8-9ed5-4995-ac57-73db5d9ff8eb)
Proposed Framework.

 - [ResNet](https://towardsdatascience.com/review-resnet-winner-of-ilsvrc-2015-image-classification-localization-detection-e39402bfa5d8)-50 is used as backbone.

>Finally, by combining phase-1 and phase-2 together, the proposed pipeline achieves 16.7 performance gain in top-1 accuracy over the supervised baseline.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/0d11a6e6-f7c3-4ba9-afdd-5a74d2110ec1)
Linear Classifier.

>The proposed margin loss is less sensitive to the number negatives and can be used in a data-deficient setting.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/498b38c6-f941-4a68-9f04-15a99b2e0be8)
Bag of Tricks.

>Several other tricks and stronger backbone models are used for better performance: Larger Resolution, [AutoAugment](https://sh-tsang.medium.com/review-autoaugment-learning-augmentation-strategies-from-data-image-classification-af27dea9a839), [ResNeXt](https://towardsdatascience.com/review-resnext-1st-runner-up-of-ilsvrc-2016-image-classification-15d7f17b42ac)-101, label-smooth ([Inception-v3](https://medium.com/@sh.tsang/review-inception-v3-1st-runner-up-image-classification-in-ilsvrc-2015-17915421f77c)), 10-Crop, and 2-model ensemble.

**Comment by NorbertZheng** [comment]
- It's worth noting: distillation > full fine-tune > linear probing > supervised training.
 - Margin contrastive loss achieves better result with smaller batch size.
 - Tricks!!!

**Comment by NorbertZheng** [comment]
## Reference
 - [2020 arXiv] [MoCo v2+Distillation] [Distilling Visual Priors from Self-Supervised Learning](https://arxiv.org/abs/2008.00261).
